daffodil-lib/src/main/scala/edu/illinois/ncsa/daffodil/xml/XMLUtils.scala

```diff
@@ -671,13 +676,27 @@ object XMLUtils {
             case xsiNilAttr @ PrefixedAttribute("xsi", "nil", Text("true"), _) if (xsiNilAttr.getNamespace(e) == null) => {
               true
             }
-            case attr => {
-              if (ns.length > 0) {
-                !ns.contains(NS(attr.getNamespace(e)))
-              } else {
-                false
-              }
+            case dafIntAttr @ PrefixedAttribute(pre, _, _, _) if (pre ne null) && (dafIntAttr.getNamespace(e) == XMLUtils.DAFFODIL_INTERNAL_NAMESPACE.toString) => {
+              Assert.invariant(pre != "")
```

Comment:
Is the only situation where pre could be an empty string is if default xmlsns was the daffodil internal namespace? Seems some like something really weird and probably dumb to do, but technically it's legal, I think? So should this assertion be removed?

Reply:
Well, dafint is ours. Not the users choice how to use it. I don't think we want to allow use of dafint as a default namespace. Certainly I'd consider it  a programming error in daffodil if dafint ever was a default namespace hence, this is an invariant.